### PR TITLE
nlohmann-json: add v3.11.2 and make 'multiple_headers' the default

### PR DESCRIPTION
**Pull request by falbrechtskirchinger**
Add new version of nlohmann-json: https://github.com/nlohmann/json/releases/tag/v3.11.2

Using the multiple-header variant is recommended because it produces more useful error messages (i.e., which subsystem fails to compile). Upstream made that the default in v3.11.0.

### CODE REVIEW

**Comment by falbrechtskirchinger** [comment]
v3.11.1 may not be recommended either. Evaluating reported regression …

**Review by tldahlgren — DISMISSED** [review]
Confirmed sha256. Deferring merge decision to give maintainer a chance to comment.

**Comment by falbrechtskirchinger** [comment]
The number of issues/regressions in v3.11.1 is stacking up. I'd hold off for now. I'm working on fixes but don't expect a release for a few days.

**Comment by tldahlgren** [comment]
> The number of issues/regressions in v3.11.1 is stacking up. I'd hold off for now. I'm working on fixes but don't expect a release for a few days.

Oops.  Just noticed this is DRAFT.

**Comment by falbrechtskirchinger** [comment]
Updated to 3.11.2.

@tldahlgren The sha256 you confirmed earlier has changed.

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256

**Comment by tldahlgren** [comment]
Oops.  Merged without waiting on maintainer.  We can revert if you want @ax3l .

**Pull request merged**
